tss-esapi/examples/certify.rs

```diff
@@ -0,0 +1,598 @@
+ *                         │                                │                       │           │           │
+ *                         │                                │                       │           ▼           │
+ *                         │                                │                       │ AIK is legitimate!!   │
+ *                         │                                │                       │                       │
+ *                         │                                │                       │                       │
+ *      Load another key  ─┼───▶ Certify Object with AIK    │                       │                       │
+ *                         │                │               │                       │                       │
+ *                         │                │               │                       │                       │
+ *                         │                ▼               │                       │                       │
+ *                         │  ┌──────────────────────────┐  │                       │                       │
+ *                         │  │                          │  │                       │   Verify Attestation  │
+ *                         │  │       Attestation        │──┼───────────────────────┼─▶    is from AIK      │
+ *                         │  │                          │  │                       │                       │
+ *                         │  └──────────────────────────┘  │                       │                       │
+ *                         └────────────────────────────────┘                       └───────────────────────┘
+ *
+ */
+
+use tss_esapi::{
+    abstraction::{
+        ak::{create_ak, load_ak},
+        cipher::Cipher,
+        ek::{create_ek_public_from_default_template, retrieve_ek_pubcert},
+        AsymmetricAlgorithmSelection,
+    },
+    attributes::{ObjectAttributesBuilder, SessionAttributesBuilder},
+    constants::SessionType,
+    handles::{AuthHandle, KeyHandle, SessionHandle},
+    interface_types::{
+        algorithm::{HashingAlgorithm, PublicAlgorithm, SignatureSchemeAlgorithm},
+        ecc::EccCurve,
+        key_bits::RsaKeyBits,
+        reserved_handles::Hierarchy,
+        session_handles::{AuthSession, PolicySession},
+    },
+    structures::{
+        CreatePrimaryKeyResult, Data, Digest, EccPoint, EccScheme, HashScheme, MaxBuffer,
+        PublicBuilder, PublicEccParametersBuilder, SignatureScheme, SymmetricCipherParameters,
+        SymmetricDefinition, SymmetricDefinitionObject,
+    },
+    traits::Marshall,
+    Context, TctiNameConf,
+};
+
+use std::convert::{TryFrom, TryInto};
+
+fn main() {
+    env_logger::init();
+    // Create a pair of TPM's contexts - It's not "perfect" but it's what we will use
+    // to represent the two TPM's in our test.
+    //
+    // It's recommended you use `TCTI=device:/dev/tpmrm0` for the linux kernel
+    // tpm resource manager.
```

Comment:
Most other examples have this too though. The intent isn't to explain the difference between tpm0 and tpmrm0, as much as to say "here is a resource managed tpm you can probably just use with out worrying".

Reply:
Right, why I mention this here is because you have two concurrent contexts, that's why you might have issues with them existing at the same time. Forgot to make this clear in my initial comment...

Reply:
Sure, I've updated the comments here in this example *and* the duplication examples which pull the same trick. :)